plugin/gcp/resources.py

```diff
@@ -0,0 +1,304 @@
+# #######
+# Copyright (c) 2014 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+from plugin.gcp import utils
+from plugin.gcp.service import GoogleCloudPlatform
+from plugin.gcp.service import GCPError
+from plugin.gcp.service import blocking
+
+
+class FirewallRule(GoogleCloudPlatform):
+    def __init__(self,
+                 config,
+                 logger,
+                 firewall,
+                 network):
+        """
+
+        :param auth:
+        :param project:
+        :param logger:
+        :param firewall: firewall dictionary with a following structure:
+        firewall = {'name': 'firewallname',
+                    'allowed: [{ 'IPProtocol': 'tcp', 'ports': ['80']}],
+                    'sourceRanges':['0.0.0.0/0'],
+                    'sourceTags':['tag'], (optional)
+                    'targetTags':['tag2'] (optional)
+                    }
+        ref. https://cloud.google.com/compute/docs/reference/latest/firewalls
+        :param network: network name the firewall rule is connected to
+        :return:
+        """
+        super(FirewallRule, self).__init__(config, logger)
+        self.firewall = firewall
+        self.network = network
+        self.firewall['name'] = self.get_name()
+        self.name = self.firewall['name']
+
+    def get_name(self):
+        """
+        Prefix firewall rule name with network name
+
+        :return: network prefixed firewall rule name
+        """
+        name = '{0}-{1}'.format(self.network, self.firewall['name'])
+        return utils.get_gcp_resource_name(name)
+
+    @blocking(True)
```

Comment:
Hi,
Yes you're totally right. I'm aware of the ctx.operation.retry API but synchronous solution is easier to have it tested and add new functionalities. This will be changed to ctx.operation.retry in 2 sprints(each is about 2 weeks long) when most of tests will be written.